cosipy/data_io/UnBinnedData.py

```diff
@@ -655,29 +656,61 @@ def combine_unbinned_data(self, input_files, output_name=None):
 
         self.cosi_dataset = {}
         counter = 0
-        for each in input_files:
-
-            logger.info("adding %s..." % each)
-    
-            # Read dict from hdf5 or fits:
-            this_dict = self.get_dict(each)
-
-            # Combine dictionaries:
-            if counter == 0:
-                for key in this_dict:
-                    self.cosi_dataset[key] = this_dict[key]
-            
-            if counter > 0:
-                for key in this_dict:
-                    self.cosi_dataset[key] = np.concatenate((self.cosi_dataset[key],this_dict[key]))
-                    
-            counter =+ 1
-            
-            # Clear unused memory:
-            gc.collect()
 
-        # Write unbinned data to file (either fits or hdf5):
-        if output_name != None:
-            self.write_unbinned_output(output_name)
+        with tempfile.TemporaryDirectory() as tmpdir:
+            for each in input_files:
+                logger.info("adding %s..." % each)
+
+                # Read dict from hdf5 or fits:
```

Comment:
Hi @krishnatejavedula, Can you please use `this_dict = self.get_dict(each)` here to replace lines 665 - 668. See my line 663 that you deleted.

Reply:
Hello @ckarwin, sorry about that. I have made the changes you've requested.